src/publish.proj

```diff
@@ -1,41 +1,52 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
-  <Import Project="$(ToolsDir)PublishContent.targets" />
-  <Import Project="$(ToolsDir)versioning.targets" />
 
-  <!-- gathers the items to be published -->
-  <Target Name="GatherItemsForPattern">
-    <Error Condition="'$(PublishPattern)' == ''" Text="Please specify a value for PublishPattern using standard msbuild 'include' syntax." />
-    <ItemGroup>
-      <ForPublishing Include="$(PublishPattern)" />
-    </ItemGroup>
-    <!-- add relative blob path metadata -->
-    <ItemGroup>
-      <ForPublishing>
-        <RelativeBlobPath Condition="'$(PublishTestNativeBins)' != 'true'">$(__BuildType)/%(RecursiveDir)%(Filename)%(Extension)</RelativeBlobPath>
-        <RelativeBlobPath Condition="'$(PublishTestNativeBins)' == 'true'">$(__BuildType)/%(RecursiveDir)%(Filename)%(Extension)</RelativeBlobPath>    
-      </ForPublishing>
-    </ItemGroup>
-    <Error Condition="'@(ForPublishing)' == ''" Text="No items were found matching pattern '$(PublishPattern)'." />
-  </Target>
+  <PropertyGroup>
+    <FeedsDir Condition="'$(BuildOS)' == 'Windows_NT'">$(MSBuildThisFileDirectory)../component/Microsoft.Dotnet.Build.Tasks.Feed/desktop/</FeedsDir>
+    <FeedsDir Condition="'$(BuildOS)' != 'Windows_NT'">$(MSBuildThisFileDirectory)../component/</FeedsDir>
+  </PropertyGroup>
+
+  <Import Project="$(FeedsDir)Publish.targets" />
+
+  <PropertyGroup>
+    <PackagesPattern Condition="'$(PackagesPattern)' == '' and '$(PublishTestNativeBins)' != 'true'">$(PackagesBinDir)pkg\*.nupkg</PackagesPattern>
```

Comment:
What does PublishTestNativeBins represent?

Reply:
On non-windows, we publish test native binaries (that are built during the product build) to an azure container, so that during the test build (on windows), we can restore those binaries and put them into Core_Root.

Reply:
Any reason that we have these as mutually exclusive instead of just always publishing them then? Also is there a reason we publish them to a different container?

Reply:
We only want to publish them on non-Windows. The different container is to avoid them being downloaded by the publish step, or downloading packages while restoring test native binaries.

Reply:
I don't see us doing this as part of our official builds. Can you point me at the publish step that does this?

Reply:
https://github.com/dotnet/coreclr/blob/master/publish-packages.sh#L23

publish-packages.sh does it by default